VocaDbWeb/App_Code/ArchivedEntryHelpers.cshtml

```diff
@@ -108,16 +108,32 @@
 			var style = (ver.AnythingChanged ? "" : "muted");
 			<tr>
 				<td>
-					@if (linkFunc != null) {
-						<a href="@linkFunc(ver.Id)" class="@style">
-							@ver.Contract.Version
-							(@Translate.EntryStatusNames[ver.Status])
+					@if (linkFunc != null && (Login.CanViewHiddenRevisions || !ver.Hidden)) {
+						<a href="@linkFunc(ver.Id)" class="@style">
+							@if (ver.Hidden) {
+								<s>
+									@ver.Contract.Version
+									(@Translate.EntryStatusNames[ver.Status])
+								</s>
+							} else {
+								<text>
+								@ver.Contract.Version
+								(@Translate.EntryStatusNames[ver.Status])
+								</text>
+							}
 						</a>
 					} else {
-						<text>
-							@ver.Contract.Version
-							(@Translate.EntryStatusNames[ver.Status])
-						</text>
+						if (ver.Hidden) {
+							<s>
```

Comment:
Instead of different tags, I would consider CSS style to differentiate how to show hidden revisions